src/common/set.c

```diff
@@ -489,9 +494,17 @@ util_poolset_parse(const char *path, int fd, struct pool_set **setp)
 		return -1;
 	}
 
+	fd = dup(fd);
+	if (fd < 0) {
+		ERR("!dup");
+		return -1;
```

Comment:
`Free(set)` is missing. Same above.
Seems like it would be better to do `goto err`, rather than just `return -1`.